test_weather_data_fetcher.py

```diff
@@ -0,0 +1,61 @@
+import os
+import dotenv
+from unittest import mock
+
+from weather_data_fetcher import fetch_location_coords, fetch_weather_data, get_api_key
+
+
+class TestWeatherDataFetcher:
+    def test_fetch_coords(self):
+        coords_berlin = fetch_location_coords("Berlin")
+        coords_difference = [sum(x) for x in zip(coords_berlin, (-52.520008, -13.404954))]
+        assert sum(list(coords_difference)) < 0.01
+
+    @mock.patch("weather_data_fetcher.fetch_new_data", autospec=True)
+    def test_fetch_data(self, request_mock):
+        request_mock.return_value = {
```

Comment:
You can make your life easier here and just put into the json payload what you extract in the end. This makes everything much much shorter.